packages/react-devtools-shared/src/__tests__/store-test.js

```diff
@@ -1008,4 +1010,197 @@ describe('Store', () => {
+    test('from react get counted', () => {
+      const container = document.createElement('div');
+      function Example() {
+        return <div data-camelCased="should be lowercase" />;
+      }
+
+      withErrorsOrWarningsIgnored(
+        ['Warning: React does not recognize the `%s` prop on a DOM element.'],
+        () => {
+          act(() => ReactDOM.render(<Example />, container));
+        },
+      );
+
+      expect(store.errorsAndWarnings).toEqual(
+        new Map([[2, {errors: 1, warnings: 0}]]),
+      );
+    });
+
+    test('can be cleared for the whole app', () => {
+      function Example() {
+        console.error('test-only: render error');
+        console.warn('test-only: render warning');
+        return null;
+      }
+      const container = document.createElement('div');
+      withErrorsOrWarningsIgnored(['test-only:'], () => {
+        act(() =>
+          ReactDOM.render(
+            <React.Fragment>
+              <Example />
+              <Example />
+            </React.Fragment>,
+            container,
+          ),
+        );
+      });
+
+      store.clearErrorsAndWarnings();
+      // flush events to the renderer
+      jest.runAllTimers();
+
+      expect(store.errorsAndWarnings).toEqual(
+        new Map([
+          [2, {errors: 0, warnings: 0}],
+          [3, {errors: 0, warnings: 0}],
+        ]),
+      );
+    });
+
+    test('can be cleared for particular Fiber (only warnings)', () => {
+      function Example() {
+        console.error('test-only: render error');
+        console.warn('test-only: render warning');
+        return null;
+      }
+      const container = document.createElement('div');
+      withErrorsOrWarningsIgnored(['test-only:'], () => {
+        act(() =>
+          ReactDOM.render(
+            <React.Fragment>
+              <Example />
+              <Example />
+            </React.Fragment>,
+            container,
+          ),
+        );
+      });
+
+      store.clearWarningsForElement(2);
+      // Flush events to the renderer.
+      jest.runAllTimers();
+
+      expect(store.errorsAndWarnings).toEqual(
+        new Map([
+          [2, {errors: 1, warnings: 0}],
+          [3, {errors: 1, warnings: 1}],
+        ]),
+      );
+    });
+
+    test('can be cleared for a particular Fiber (only errors)', () => {
+      function Example() {
+        console.error('test-only: render error');
+        console.warn('test-only: render warning');
+        return null;
+      }
+      const container = document.createElement('div');
+      withErrorsOrWarningsIgnored(['test-only:'], () => {
+        act(() =>
+          ReactDOM.render(
+            <React.Fragment>
+              <Example />
+              <Example />
+            </React.Fragment>,
+            container,
+          ),
+        );
+      });
+
+      store.clearErrorsForElement(3);
```

Comment:
Typo, I think. I'll fix.

Reply:
This was actually intentional to test that it doesn't just clear the errors/warnings for the first Fiber. Should've added a comment explaining that these not only test what type is cleared but also that the given ID has an effect.

Though it's simpler to just add another action+assertion after that i.e. first clear ID 3, assert, clear ID 2, assert in both tests.

Reply:
Oh. We didn't have a test in this file for the clear-errors (only clear-warnings) so I assumed this was accidental.